app/src/main/java/com/example/android/oscarine/signin/SigninViewModel.kt

```diff
@@ -21,14 +25,19 @@ class SigninViewModel: ViewModel() {
 
 
     private fun postForSigningInUser(userDetails: LoginUser) {
-        OscarineApi.retrofitService.loginUser(userDetails).enqueue(object: Callback<Any> {
-            override fun onResponse(call: Call<Any>, response: Response<Any>) {
-                _response.value = response.code().toString()
-                resetValuesOnSigninSuccess()
+        OscarineApi.retrofitService.loginUser(userDetails).enqueue(object: Callback<LoginServerResponse> {
+            override fun onFailure(call: Call<LoginServerResponse>, t: Throwable) {
+                _response.value = "Failure: " + t.message
             }
 
-            override fun onFailure(call: Call<Any>, t: Throwable) {
-                _response.value = "Failure: " + t.message
+            override fun onResponse(
+                call: Call<LoginServerResponse>,
+                response: Response<LoginServerResponse>
+            ) {
+                _response.value = response.code().toString()
+                var token = response.body()?.access_token ?: ""
```

Comment:
val

And don't use ?.

Reply:
`Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type LoginServerResponse?`

Reply:
Use !!